src/frontend/index.js

```diff
@@ -111,34 +107,25 @@ function init() {
 			}
 			
 			persistentConnection.on('connected', function (data) {
-				chunkParser.reset();
 				jsonPerLineParser.reset();
 			});
 			persistentConnection.on('data', function (data) {
-				chunkParser.write(data);
+				jsonPerLineParser.write(data);
 			});
 			persistentConnection.on('end', reconnectOnError);
 			persistentConnection.on('error', reconnectOnError);
 			
-			chunkParser.on('data', function (data) {
-				jsonPerLineParser.write(data);
-			});
-			chunkParser.on('end', reconnectOnError);
-			chunkParser.on('error', reconnectOnError);
-			
 			jsonPerLineParser.on('data', function (data) {
-				data = (data && data.value0 && data.value0.data);
-				
-				if (data && data.length > 0) {
+				if (data && typeof data.x === 'number' && typeof data.y === 'number') {
 					// Advance the time that will go in the next request
 					// to the latest data sample time (add 1 to avoid duplicates):
-					queryParams.since = data[data.length-1].x + 1;
+					queryParams.since = data.x + 1;
 					persistentConnection.setQuery(queryParams);
 					
 					// Notify that the data has been received:
 					dispatcher.emit('receive-data', {
 						dataUrl: dataUrl,
-						data: data
+						data: [ data ]
```

Comment:
I've implemented receiving data as multiple points to be able to push them in batches if needed. It's been used this way before the rewrite to the C++ backend which pushes only the points, one per chunk, not some data with an array of points inside. This is not related to caching at this level, it's batch send/receive.